### PR TITLE
feat(phone): Add route for /recovery-phone/send_code

### PR DESCRIPTION
**Pull request by vbudhram**
## Because

- We want to send a code to a recovery phone to verify it

## This pull request

- Adds a new route `/recovery-phone/send_code`

## Issue that this pull request solves

Closes: https://mozilla-hub.atlassian.net/browse/FXA-10357

## Checklist

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

@dschom Tagged you for review since this will use some of the Glean metrics in https://github.com/mozilla/fxa/pull/18067

### CODE REVIEW

**Comment by vbudhram** [comment]
@dschom This should be ready now

File: packages/fxa-auth-server/lib/routes/recovery-phone.ts [hunk 1]
```diff
@@ -129,6 +160,18 @@ export const recoveryPhoneRoutes = (
         return recoveryPhoneHandler.confirm(request);
       },
     },
+    {
+      method: 'POST',
+      path: '/recovery-phone/send_code',
+      options: {
```
Comment (dschom) [inline packages/fxa-auth-server/lib/routes/recovery-phone.ts:166]: Do we need a validate check on the phoneNumber being sent in the payload?

**Review by dschom — APPROVED** [review]
LGTM. Please take a look at the nit about handler validation.

File: packages/fxa-auth-server/lib/routes/recovery-phone.ts (hunk 1 above)
Comment (vbudhram) [reply packages/fxa-auth-server/lib/routes/recovery-phone.ts:166]: @dschom Don't believe so, this api should send a code to the phone number created by the create api.

Comment (dschom) [reply packages/fxa-auth-server/lib/routes/recovery-phone.ts:166]: Sorry, that was a misread on my part.

**Pull request merged**